src/joystick/virtual/SDL_virtualjoystick.c

```diff
@@ -142,6 +142,10 @@ SDL_JoystickID SDL_JoystickAttachVirtualInner(const SDL_VirtualJoystickDesc *des
         SDL_InvalidParamError("desc");
         return 0;
     }
+    if (desc->version != sizeof(*desc)) {
```

Comment:
Internally, we'll keep the old version structs around so we can compare sizes, I assume. Since there's only one version here, refusing to work if the size doesn't match is fine, but in the future we'll want to manage the differences vs an older version internally.

Also, is there ever a time a compiler might pad out the end of a struct unexpectedly?

Reply:
> Internally, we'll keep the old version structs around so we can compare sizes, I assume. Since there's only one version here, refusing to work if the size doesn't match is fine, but in the future we'll want to manage the differences vs an older version internally.

Yup, that's right.
 
> Also, is there ever a time a compiler might pad out the end of a struct unexpectedly?

Possibly. I'll add a compile time assert to catch that.